Mathlib/MeasureTheory/Measure/MeasureSpace.lean

```diff
@@ -513,6 +513,24 @@ theorem measure_iUnion_eq_iSup [Countable ι] {s : ι → Set α} (hd : Directed
         _ ≤ ⨆ n, μ (t n) := le_iSup (μ ∘ t) N
 #align measure_theory.measure_Union_eq_supr MeasureTheory.measure_iUnion_eq_iSup
 
+/-- Continuity from below: the measure of the union of a sequence of
+(not necessarily measurable) sets is the supremum of the measures of the partial unions. -/
+theorem measure_iUnion_eq_iSup' (r : ι → ι → Prop) (r_trans : Transitive r) [Countable ι]
```

Comment:
I am not completely convinced by the added generality here, because I don't imagine we will apply this version of the lemma. On the other hand, I am completely convinced by `tendsto_measure_iUnion'`. What about specializing `measure_iUnion_eq_iSup'` to take a directed order instead of a general relation, to make it easier to parse while not really changing the applicability?

Same thing for the decreasing version below.

Reply:
I agree the hypotheses are not very natural, a directed preorder is of course enough. I also discovered [Set.Accumulate](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Accumulate.html#Set.Accumulate) which simplifies the proofs involving unions.